cloudsmith/resource_saml_auth.go

```diff
@@ -0,0 +1,257 @@
+	}
+
+	d.SetId(generateSAMLAuthID(organization, result))
+	return samlAuthRead(ctx, d, m)
+}
+
+// samlAuthRead retrieves the current SAML authentication configuration
+func samlAuthRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	pc := m.(*providerConfig)
+	organization := d.Get("organization").(string)
+
+	samlAuth, resp, err := pc.APIClient.OrgsApi.OrgsSamlAuthenticationRead(pc.Auth, organization).Execute()
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(handleSAMLAuthError(err, resp, "reading SAML authentication"))
+	}
+
+	d.Set("organization", organization)
+	d.SetId(generateSAMLAuthID(organization, samlAuth))
+
+	if err := setSAMLAuthFields(d, organization, samlAuth); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+// samlAuthUpdate modifies an existing SAML authentication configuration
+func samlAuthUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	pc := m.(*providerConfig)
+	organization := d.Get("organization").(string)
+
+	samlAuth, err := buildSAMLAuthPatch(d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error building SAML auth request: %w", err))
+	}
+
+	req := pc.APIClient.OrgsApi.OrgsSamlAuthenticationPartialUpdate(pc.Auth, organization).Data(*samlAuth)
+	_, resp, err := pc.APIClient.OrgsApi.OrgsSamlAuthenticationPartialUpdateExecute(req)
+	if err != nil {
+		return diag.FromErr(handleSAMLAuthError(err, resp, "updating SAML authentication"))
+	}
+
+	return samlAuthRead(ctx, d, m)
+}
+
+// samlAuthDelete disables SAML authentication for the organization
+func samlAuthDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	pc := m.(*providerConfig)
+	organization := d.Get("organization").(string)
+
+	samlAuth := cloudsmith.NewOrganizationSAMLAuthRequestPatch()
+	samlAuth.SetSamlAuthEnabled(false)
+	samlAuth.SetSamlAuthEnforced(false)
+	samlAuth.SetSamlMetadataInline("")
+	samlAuth.SetSamlMetadataUrl("")
+
+	req := pc.APIClient.OrgsApi.OrgsSamlAuthenticationPartialUpdate(pc.Auth, organization).Data(*samlAuth)
+	_, resp, err := pc.APIClient.OrgsApi.OrgsSamlAuthenticationPartialUpdateExecute(req)
+	if err != nil {
+		return diag.FromErr(handleSAMLAuthError(err, resp, "deleting SAML authentication"))
+	}
+
+	d.SetId("")
+	return nil
+}
+
+// samlAuthImport handles importing existing SAML authentication configurations
+func samlAuthImport(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	pc := m.(*providerConfig)
+	organization := d.Id()
+
+	samlAuth, resp, err := pc.APIClient.OrgsApi.OrgsSamlAuthenticationRead(pc.Auth, organization).Execute()
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			return nil, fmt.Errorf("SAML authentication not found for organization %s", organization)
```

Comment:
NAB: At _some_ point, we'll need to reconcile the org/namespace/workspace nomenclature, so that it's just workspace. :not-today: